### PR TITLE
Gnc enhancements

### PR DESCRIPTION
**Pull request by lucacarlone**
- [x] added interface to specify known outliers in GNC
- [x] added example with GNC

### CODE REVIEW

**Comment by lucacarlone** [comment]
@dellaert : no hurry! 
@jingnanshi : I mostly added extra interfaces the user can use. They are well tested, but let me know if you see something off or you have speedups to suggest

File: gtsam/nonlinear/GncOptimizer.h [hunk 1]
```diff
@@ -350,24 +403,28 @@ class GncOptimizer {
...
-                        params_.knownInliers.end(),
+                        knownWeights.begin(), knownWeights.end(),
                         std::inserter(unknownWeights, unknownWeights.begin()));
 
     // update weights of known inlier/outlier measurements
     switch (params_.lossType) {
       case GncLossType::GM: {  // use eq (12) in GNC paper
         for (size_t k : unknownWeights) {
-          if (nfg_[k]) {
+           if (nfg_[k]) {
```
Comment (jingnanshi) [inline gtsam/nonlinear/GncOptimizer.h:427]: nitpick: there's an extra space here

**Review by jingnanshi — APPROVED** [review]
:+1: Left one formatting comment.

**Comment by lucacarlone** [comment]
@jingnanshi  - thanks! 
@dellaert : this should be ready to go.

**Comment by dellaert** [comment]
Awesomesauce! will merge.

**Pull request merged**